From 56bbf73327cd1b39c947863eff4863805c008b8c Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Tue, 22 Nov 2022 13:51:08 +0100
Subject: [PATCH 5/5] accton-as4610-54te: do not call
 hwmon_device_register_with_info() with NULL chip

hwmon_device_register_with_info() is supposed to require hwmon_chip_info
being passed, so passing NULL is not valid.

Recent kernels enforce this, but supplying valid hwmon_chip_info would
require rewriting the whole driver.

So instead revert to the old call and live with the warning.

// <smpl>
@@
expression ret;
expression E1,E2;
@@
-ret = hwmon_device_register_with_info(E1, E2, NULL, NULL, NULL);
+ret = hwmon_device_register(E1);
// </smpl>

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 .../as4630-54te/modules/builds/x86-64-accton-as4630-54te-psu.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/packages/platforms/accton/x86-64/as4630-54te/modules/builds/x86-64-accton-as4630-54te-psu.c b/packages/platforms/accton/x86-64/as4630-54te/modules/builds/x86-64-accton-as4630-54te-psu.c
index f39985560a46..33c21af3e478 100644
--- a/packages/platforms/accton/x86-64/as4630-54te/modules/builds/x86-64-accton-as4630-54te-psu.c
+++ b/packages/platforms/accton/x86-64/as4630-54te/modules/builds/x86-64-accton-as4630-54te-psu.c
@@ -183,8 +183,7 @@ static int as4630_54te_psu_probe(struct i2c_client *client,
 		goto exit_free;
 	}
 
-	data->hwmon_dev = hwmon_device_register_with_info(&client->dev,
-									"as4630_54te_psu", NULL, NULL, NULL);
+	data->hwmon_dev = hwmon_device_register(&client->dev);
 	if (IS_ERR(data->hwmon_dev)) {
 		status = PTR_ERR(data->hwmon_dev);
 		goto exit_remove;
-- 
2.38.1

